apps/passport-client/components/screens/EdgeCityScreens/BalancesTab.tsx

```diff
@@ -0,0 +1,234 @@
+import {
+  EdgeCityScore,
+  FrogCryptoScore,
+  HAT_TOKEN_NAME,
+  TOTAL_SUPPLY
+} from "@pcd/passport-interface";
+import _ from "lodash";
+import { useCallback, useState } from "react";
+import styled from "styled-components";
+import { useDispatch } from "../../../src/appHooks";
+import { RippleLoader } from "../../core/RippleLoader";
+import { ActionButton } from "../FrogScreens/Button";
+import { useUsernameGenerator } from "../FrogScreens/useUsername";
+
+const length = 39;
+const MOCK_DATA: EdgeCityScore[] = Array.from({ length }, (_, index) => ({
+  semaphore_id_hash: index.toString(),
+  has_telegram_username: false,
+  score: length - index, // Assuming score is a random number between 0 and 100
+  rank: index
+}));
+
+/**
+ * The Score tab shows the user their score and the leaderboard.
+ */
+export function BalancesTab({
+  score,
+  refreshScore
+}: {
+  score?: FrogCryptoScore;
+  refreshScore: () => Promise<void>;
+}): JSX.Element {
+  const [scores, setScores] = useState<EdgeCityScore[]>(MOCK_DATA);
+  const refreshScores = useCallback(async () => {
+    setScores(MOCK_DATA);
+    // requestFrogCryptoGetScoreboard(appConfig.zupassServer).then((res) => {
+    //   setScores(res.value || []);
+    // });
+  }, []);
+  // useEffect(() => {
+  //   refreshScores();
+  // }, [refreshScores]);
+
+  const getUsername = useUsernameGenerator();
+
+  if (!score || !getUsername) {
+    return <RippleLoader />;
+  }
+
+  return (
+    <Container>
+      {
+        // only show share button if user has a telegram username
```

Comment:
is the telegram stuff going to make it in tonight? we could push that to later in the week if necessary.